io/include/detray/io/backend/material_map_reader.hpp

```diff
@@ -78,31 +87,62 @@ class material_map_reader {
             auto mat_factory = std::make_shared<
                 material_map_factory<detector_t, bin_index_type>>();
 
+            DETRAY_DEBUG("Converting " << mat_grids.size()
+                                       << " material grids");
+
             // Convert the material grid of each surface
-            for (const auto &grid_data : mat_grids) {
+            for (const auto &[idx, grid_data] :
+                 detray::views::enumerate(mat_grids)) {
+
+                DETRAY_DEBUG("- grid idx=" << idx);
```

Comment:
It might be more informative to print the owner link here, i.e. the surface the material belongs to

Reply:
like so?

```suggestion
                DETRAY_DEBUG("- grid link.index=" << grid_data.grid_link.index);
```